armi/reactor/blocks.py

```diff
@@ -2136,9 +2136,13 @@ def _rotatePins(self, rotNum, justCompute=False):
                 "Cannot rotate {0} to rotNum {1}. Must be 0-5. ".format(self, rotNum)
             )
 
-        # Pin numbers start at 1. Number of pins in the block is assumed to be based on
-        # cladding count.
-        numPins = self.getNumComponents(Flags.CLAD)
+        numPins = self.getNumPins()
+        hexRings = [3 * r * (r - 1) + 1 for r in range(1, 11)]
+        for i in range(len(hexRings) - 1):
+            if numPins > hexRings[i] and numPins < hexRings[i + 1]:
+                # round numPins up
+                numPins = hexRings[i + 1]
```

Comment:
So this is trying to find the number of pins in a full hex lattice in the event we have fewer pins? So if I have two rings of pins with the center spot missing, `numPins` would be six, but this logic would round up to seven for two full rings?

If that's the case, could these two functions be helpful?

https://github.com/terrapower/armi/blob/f1f3dbea05c2564404a03d8e053f72c2af15daa9/armi/utils/hexagon.py#L107-L111

https://github.com/terrapower/armi/blob/f1f3dbea05c2564404a03d8e053f72c2af15daa9/armi/utils/hexagon.py#L150-L151

like
```python
numPins = self.getNumPins()
numRings = numRingsToHoldNumCells(numPins)
fullNumPins = totalPositionsUpToRing(numRings)
```